File: src/812.py
import sys
import logging
from typing import List, Optional
from collections import defaultdict, deque, Counter
from functools import cache, lru_cache
from bisect import bisect_left, bisect_right
from itertools import accumulate, combinations
from heapq import heapify, heappush, heappop


def logging_setting():
    LOG_LEVEL = logging.INFO    # DEBUG, INFO, WARNING, ERROR, CRITICAL
    if __debug__:
        LOG_LEVEL = logging.DEBUG

    logging.basicConfig(
        stream=sys.stderr,
        level=LOG_LEVEL,
        format="%(levelname)-6s %(asctime)s %(module)s:%(funcName)s:%(lineno)-4s %(message)s",
        datefmt='%Y/%m/%d %H:%M:%S')


class Solution:
    def area(self, p: int, q: int, r: int) -> float:
        retVal = .5 * abs((p[0]*q[1]) + (q[0]*r[1]) + (r[0]*p[1]) - (p[1]*q[0]) - (q[1]*r[0]) - (r[1]*p[0]))

        return retVal

    def largestTriangleArea(self, points: List[List[int]]) -> float:
        retVal = 0

        for triangle in combinations(points, 3):
            retVal = max(retVal, self.area(*triangle))

        return retVal


if __name__ == "__main__":
    logging_setting()

    try:
        logging.info("sys.version: %s", sys.version)
        print()

        pSolution = Solution()
        for points in [[[0, 0], [0, 1], [1, 0], [0, 2], [2, 0]], [[1, 0], [0, 0], [0, 1]]]:
            # /* Example
            #  *  Input: points = [[0,0],[0,1],[1,0],[0,2],[2,0]]
            #  *  Output: 2.00000
            #  *
            #  *  Input: points = [[1,0],[0,0],[0,1]]
            #  *  Output: 0.50000
            #  */
            logging.info("Input: points = %s", points)

            retVal = pSolution.largestTriangleArea(points)
            logging.info("Output: %.5f", retVal)

            print()
    except KeyboardInterrupt as exception:
        logging.error("%s: %s", exception.__class__.__name__, exception, exc_info=True)
        pass
    except Exception as exception:
        logging.error("%s: %s", exception.__class__.__name__, exception, exc_info=True)
        pass

    sys.exit(0)

File: src/812.cpp
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

class Solution {
   private:
    double triangleArea(int x1, int y1, int x2, int y2, int x3, int y3) {
        double retVal = 0.5 * abs((x1 * y2) + (x2 * y3) + (x3 * y1) - (x1 * y3) - (x2 * y1) - (x3 * y2));

        return retVal;
    }

   public:
    double largestTriangleArea(vector<vector<int>>& points) {
        double retVal = 0;

        int pointsSize = points.size();
        for (int i = 0; i < pointsSize; i++) {
            for (int j = i + 1; j < pointsSize; j++) {
                for (int k = j + 1; k < pointsSize; k++) {
                    retVal = max(retVal, triangleArea(points[i][0], points[i][1], points[j][0], points[j][1],
                                                      points[k][0], points[k][1]));
                }
            }
        }

        return retVal;
    }
};

int main(int argc, char** argv) {
    struct subject {
        vector<vector<int>> points;
    };
    vector<subject> testData{{{{0, 0}, {0, 1}, {1, 0}, {0, 2}, {2, 0}}}, {{{1, 0}, {0, 0}, {0, 1}}}};
    int numberOfTestCase = testData.size();
    /* Example
     *  Input: points = [[0,0],[0,1],[1,0],[0,2],[2,0]]
     *  Output: 2.00000
     *
     *  Input: points = [[1,0],[0,0],[0,1]]
     *  Output: 0.50000
     */

    Solution cSolution;
    double answer = 0;
    for (int i = 0; i < numberOfTestCase; ++i) {
        cout << "Input: points = [";
        for (long unsigned int j = 0; j < testData[i].points.size(); ++j) {
            cout << ((j == 0) ? "" : ",") << "[";
            for (long unsigned int k = 0; k < testData[i].points[j].size(); ++k) {
                cout << ((k == 0) ? "" : ",") << testData[i].points[j][k];
            }
            cout << "]";
        }
        cout << "]\n";

        answer = cSolution.largestTriangleArea(testData[i].points);
        cout.setf(ios::fixed);
        cout.precision(5);
        cout << "Output: " << answer << "\n";

        cout << "\n";
    }

    return EXIT_SUCCESS;
}
